layouts/shortcodes/include-remote-md.html

```diff
@@ -0,0 +1,26 @@
+{{/* Fetch a remote markdown file  and include it in the page. If the file has a frontmatter, define the marker of the frontmatter in the second parameter (defaults to "---"). If the content after the frontmatter includes the marker (for example, includes "---" as part of a markdown-formatted table) the content included will be incomplete. */}}
+{{ $url := .Get 0 }}
+{{ $marker := .Get 1 | default "---" }}
+
+{{/* Do not change the indentation of the following block */}}
+{{ with resources.GetRemote $url }}
+    {{ with .Err }}
+    {{ errorf "%s" . }}
+{{ else }}
+{{/*  Test for frontmatter */}}
+{{ if hasPrefix .Content $marker }}
+{{ $split := split .Content $marker }}
+
+{{ with strings.Contains .Content $marker }} {{- warnf "Remote snippet includes frontmatter marker, content is truncated: %s" $url -}}
+{{ end }}
+
+{{/*  Output stuff after the frontmatter if a frontmatter was detected. */}}
+{{- index $split 2 | markdownify | safeHTML -}}
```

Comment:
This seems to miss a few things... it would be great to find a solution here that preserves all built-in Docsy behaviors, such as:

- adding self-links
- building the in-page ToC
- indexing?

I've tried `$.Page.RenderString` instead of `markdown | safeHTML` but that seems to do about the same thing. My hugo-fu doesn't go much beyond that.

Reply:
It seems simpler is better here. I've removed all of the filters after finding https://discourse.gohugo.io/t/extend-tableofcontents-from-within-shortcode/31994/3 and it seems to work much better.

I also rewrote the split method so that I believe it only splits on the first delimiter, then rejoins the rest of the content after.

Since my changes have gotten kind of large, I've added the whole of the shortcode in the new review comment.